packages/api/client.ts

```diff
@@ -352,11 +352,25 @@ export type ServiceStopsQueryInput = {
     busStopTypes?: BusStopType[];
     stopTypes?: string[];
     adminAreaCodes?: string[];
+    useTracks?: boolean;
 };
 
-export const getServiceStops = async (dbClient: Kysely<Database>, input: ServiceStopsQueryInput) => {
+export const getServiceStops = async (
+    dbClient: Kysely<Database>,
+    input: ServiceStopsQueryInput,
+): Promise<ServiceStops | ServiceTracks> => {
     logger.info("Starting getServiceStops...");
 
+    const tracks = await dbClient
+        .selectFrom("tracks")
+        .select(["operatorServiceId as serviceId", "longitude", "latitude"])
+        .where("tracks.operatorServiceId", "=", input.serviceId)
+        .execute();
+
+    if (tracks && tracks.length > 0 && input.useTracks) {
```

Comment:
We could wrap the previous query in a check for `input.useTracks`, that way it won't run the query if not needed

Reply:
done